build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -665,12 +667,26 @@ function addPrettier(json: Record<string, any>) {
         "prettier:fix",
     );
 
-    const hasFormatScriptResolver = Object.prototype.hasOwnProperty.call(json.scripts, "format");
+    if (hasFormatScriptResolver) {
+        if (!json["scripts"]["format"].includes("lerna")) {
+            json["scripts"]["format"] = "npm run prettier:fix";
+        }
+    }
 
-    if (hasPrettierScriptResolver || hasPrettierFixScriptResolver || hasFormatScriptResolver) {
-        json["scripts"]["format"] = "npm run prettier:fix";
-        json["scripts"]["prettier"] = "prettier --check .";
-        json["scripts"]["prettier:"] = "prettier --write .";
+    if (hasPrettierScriptResolver) {
+        if (!json["scripts"]["prettier"].includes("--ignore-path")) {
+            json["scripts"]["prettier"] = "prettier --check .";
+            console.warn("Specify the path of this package.json");
+            console.warn("prettier --check . path")
```

Comment:
I don't understand what this warning means.

Reply:
Unnecessary warning removed.